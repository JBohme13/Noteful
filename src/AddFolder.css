#form-container {
    display: flex;
    flex-direction: column;
}

#folder-name {
    order: 1;
    flex: 1;
    height: 30px;
    width: 55%;
    color: darkslategrey;
    margin: 5px;
}

#folder-submit {
    order: 4;
    height: 22px;
    margin: 5px;
    border: 1px solid grey;
    border-radius: 2px;
    box-shadow: 1px 1px;
    font-size: 14px;
    color: darkslategrey;
}

#folder-submit:hover {
    box-shadow: -1px -1px;
    cursor: pointer;
}

#folder-clear {
    position: relative;
    left: 72px;
    bottom: 32px;
    height: 22px;
    margin: 5px;
    border: 1px solid grey;
    border-radius: 2px;
    box-shadow: 1px 1px;
    font-size: 14px;
    color: darkslategrey;
}

#folder-clear:hover {
    box-shadow: -1px -1px;
    cursor: pointer;
}

label {
    color: darkslategrey;
}

h2 {
    color: darkslategrey;
    font-weight: normal;
}