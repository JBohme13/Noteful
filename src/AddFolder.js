import React, { Component } from 'react'
import NotefulContext from './NotefulContext'
import './AddFolder.css'

export default class AddFolder extends Component {
    static contextType = NotefulContext;

    render() {
      const value = this.context;
      return(
          <div id='add-folder-container'>
            <form id='form-container'>
                <h2>Add new folder</h2>
                <div id='form-group'>
                  <label htmlFor='folder-name'>Folder Name</label><br/>
                  <input 
                    type='text'
                    name='folder-name'
                    id='folder-name'
                    required='required'
                  />
                  <br/>
                  <button 
                    id='folder-submit'
                    type='submit'
                    onClick={event => value.addFolder(event, document.getElementById('folder-name').value)}
                  >
                      Submit
                  </button>
                </div>
            </form>
            <button 
                id='folder-clear'
                onClick={event => value.handleClearButton(event)}
            >
                Clear
            </button>
          </div>
        )
    }
}