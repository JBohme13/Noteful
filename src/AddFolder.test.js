import React from 'react';
import ReactDOM from 'react-dom';
import AddFolder from './AddFolder';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('AddFolder component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddFolder />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<AddFolder />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('adds new folder and routes to home page', () => {
    const wrapper = shallow(<AddFolder />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});