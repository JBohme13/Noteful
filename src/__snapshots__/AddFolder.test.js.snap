// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder component adds new folder and routes to home page 1`] = `
<div
  id="add-folder-container"
>
  <form
    id="form-container"
  >
    <h2>
      Add new folder
    </h2>
    <div
      id="form-group"
    >
      <label
        htmlFor="folder-name"
      >
        Folder Name
      </label>
      <br />
      <input
        id="folder-name"
        name="folder-name"
        required="required"
        type="text"
      />
      <br />
      <button
        id="folder-submit"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
  <button
    id="folder-clear"
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;

exports[`AddFolder component renders the UI as expected 1`] = `
<div
  id="add-folder-container"
>
  <form
    id="form-container"
  >
    <h2>
      Add new folder
    </h2>
    <div
      id="form-group"
    >
      <label
        htmlFor="folder-name"
      >
        Folder Name
      </label>
      <br />
      <input
        id="folder-name"
        name="folder-name"
        required="required"
        type="text"
      />
      <br />
      <button
        id="folder-submit"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
  <button
    id="folder-clear"
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;
